pr-dhl-woocommerce/pr-dhl-woocommerce.php

```diff
@@ -114,6 +114,12 @@ class PR_DHL_WC {
 	public function __construct() {
 		// add_action( 'init', array( $this, 'init' ) );
 		// add_action( 'plugins_loaded', array( $this, 'init' ) );
+		if ( class_exists( 'WC_Shipping_Method' ) ) {
+			$this->define_constants();
+		} else {
+			// Throw an admin error informing the user this plugin needs WooCommerce to function
+			add_action( 'admin_notices', array( $this, 'notice_wc_required' ) );
+		}
         add_action( 'init', array( $this, 'load_plugin' ), 0 );
 		add_action( 'before_woocommerce_init', array( $this, 'declare_wc_hpos_compatibility' ), 10 );
```

Comment:
```suggestion
		if ( class_exists( 'WC_Shipping_Method' ) ) {
			$this->define_constants();
			add_action( 'init', array( $this, 'load_plugin' ), 0 );
			add_action( 'before_woocommerce_init', array( $this, 'declare_wc_hpos_compatibility' ), 10 );
		} else {
			// Throw an admin error informing the user this plugin needs WooCommerce to function
			add_action( 'admin_notices', array( $this, 'notice_wc_required' ) );
		}
```
What do you think if we change move the other `add_action()` code to be inside `if ( class_exists( 'WC_Shipping_Method' ) ) { ... }`. 

So we dont need to use that condition again under `load_plugin()`:
```
public function load_plugin() {
			$this->base_country_code = $this->get_base_country();
			$this->includes();
			$this->init_hooks();
}
```